### PR TITLE
Support for python3.9 and 3.10 added

### PR DESCRIPTION
**Pull request by Chandrahas-B**
#1 

> Currently only supporting Python 3.11 because that's all I've tested on, but supporting 3.9+ should be easy enough.
> 
> To do:
> 
> * [x]  Set up testing matrix (tox or GitHub CI)
> * [x]  Fix any potential issues with Python 3.9 and 3.10

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (Chandrahas-B) [inline .github/workflows/ci.yml:1]: Added matrix support

File: pyproject.toml
Comment (Chandrahas-B) [inline pyproject.toml:1]: Generated the poetry lock with the modified pyproject.toml file

**Review by Chandrahas-B — COMMENTED** [review]
Tested with both 3.9 and 3.10 (in a conda environment)

**Comment by leonoverweel** [comment]
Thanks for the PR @Chandrahas-B! Looks great, merging :)

**Pull request merged**